apps/dokuha-roadmap-backend/src/index.ts

```diff
@@ -49,4 +53,261 @@ app.post("/users", async (c) => {
   }
 });
 
+/*****************************************
+ * create learning content
+ *****************************************/
+const createLearningContentSchema = z.object({
+  userId: z.string().uuid(),
+  title: z.string().min(1).max(255),
+  totalPage: z.number().min(1),
+  currentPage: z.number().min(1).optional(),
```

Comment:
[IMO]
zodのエラーメッセージをカスタマイズしたほうが良いと思ったのですがいかがでしょう
※もし以下コードで問題なければupdateにも同様のバリデーションがあるので、あわせて直してくれるとうれしみ

```suggestion
  title: z.
    string().
    min(1, { message: "タイトルは1文字以上で入力してください" }).
    max(255, { message: "タイトルは255文字以下で入力してください" }),
  totalPage: z.
    number().
    min(1, { message: "ページ数は1以上の値を入力してください"  }),
  currentPage: z.
    number().
    min(1, { message: "ページ数は1以上の値を入力してください"  }).
    optional(),
```

Reply:
@k-naoyuki 
確かに！！

ただ今回の場合は不要かなと思います！
というのも、minやmaxなどの比較的軽微なバリデーションエラーであれば、実はライブラリ側で以下のようなエラーメッセージを出してくれるためです😎

絶対に日本語のエラーメッセージしたい...!的なこだわりがなければ、全部の箇所に同じ定数を埋める方が若干冗長かなと！


#### タイトルが 255 文字を超えていた例
```
% curl -X POST http://localhost:8787/learning-contents -H "Content-Type: application/json" -d '{
    "userId": "d1335a98-684e-452d-ab64-9a808f2690a1",
    "title": "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456",
    "totalPage": 300,
    "currentPage": 1,
    "note": "初めてのプログラミング学習"
  }'
{"success":false,"error":[{"code":"too_big","maximum":255,"type":"string","inclusive":true,"exact":false,"message":"String must contain at most 255 character(s)","path":["title"]}]}% 
```

#### currentPageが1より小さかったとき
```
% curl -X POST http://localhost:8787/learning-contents -H "Content-Type: application/json" -d '{
    "userId": "d1335a98-684e-452d-ab64-9a808f2690a1",
    "title": "123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345",
    "totalPage": 300,
    "currentPage": 0,
    "note": "初めてのプログラミング学習"
  }'
{"success":false,"error":[{"code":"too_small","minimum":1,"type":"number","inclusive":true,"exact":false,"message":"Number must be greater than or equal to 1","path":["currentPage"]}]}% 
```


#### 今回あえてメッセージを追加している「data.currentPage <= data.totalPage」の方は、メッセージを入れないと以下みたいな感じになって何の問題だったのかがわかりにくいので入れている
```
% curl -X POST http://localhost:8787/learning-contents -H "Content-Type: application/json" -d '{
    "userId": "d1335a98-684e-452d-ab64-9a808f2690a1",
    "title": "1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345",
    "totalPage": 300,
    "currentPage": 301,
    "note": "初めてのプログラミング学習"
  }'
{"success":false,"error":[{"code":"too_big","maximum":255,"type":"string","inclusive":true,"exact":false,"message":"String must contain at most 255 character(s)","path":["title"]},{"code":"custom","message":"Invalid input","path":[]}]}% 
```

カスタムメッセージを入れるとこんな感じ
```
% curl -X POST http://localhost:8787/learning-contents -H "Content-Type: application/json" -d '{
    "userId": "d1335a98-684e-452d-ab64-9a808f2690a1",
    "title": "1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345",
    "totalPage": 300,
    "currentPage": 301,
    "note": "初めてのプログラミング学習"
  }'
{"success":false,"error":[{"code":"too_big","maximum":255,"type":"string","inclusive":true,"exact":false,"message":"String must contain at most 255 character(s)","path":["title"]},{"code":"custom","message":"現在のページは総ページ数を超えることはできません","path":[]}]}%
```

Reply:
@ittyi 
zod側でメッセージ出てるのは確認したうえでの米でした！（単にコメントしてみたかったっていう。。）

>全部の箇所に同じ定数を埋める方が若干冗長かなと〜

そうね、i18nとかnext-intlとか使うとキレイにできえうぽいけど、そもそもそこまでは不要かなーと思いました

Reply:
お〜
これイケてる。。
https://zenn.dev/aiji42/articles/171f26af4e5b5c#zod-%2B-i18next

ちょうどこの zod 定義フロントエンドでも共通で使えるのではと考えているので、フロントのフォーム入力を実装する時とか（しばらく後の想定）に入れてみるといいのかもしれないね🤔